examples/fluids/shallow-water/qfunctions/setup_geo.h

```diff
@@ -77,125 +77,92 @@ CEED_QFUNCTION(SetupGeo)(void *ctx, CeedInt Q,
   CeedScalar (*qdata)[CEED_Q_VLA] = (CeedScalar(*)[CEED_Q_VLA])out[0];
   // *INDENT-ON*
 
+  CeedInt panel = -1;
+
   CeedPragmaSIMD
-  // Quadrature Point Loop to determine on which panel the element belongs to
+  // Quadrature point loop to determine which panel the element belongs to
   for (CeedInt i=0; i<Q; i++) {
-    // Read local panel coordinates and relative panel index
-    const CeedScalar x[2] = {X[0][i],
-                             X[1][i]
-                            };
     const CeedScalar pidx = X[2][i];
-    if (pidx )
+    if (pidx != -1)
+      panel = pidx;
     break;
   }
 
+  // Check that we found nodes panel
+  if (panel == -1)
+    return -1;
+
   CeedPragmaSIMD
-  // Quadrature Point Loop for metric factors
+  // Quadrature point loop for metric factors
   for (CeedInt i=0; i<Q; i++) {
     // Read local panel coordinates and relative panel index
-    const CeedScalar x[2] = {X[0][i],
-                             X[1][i]
-                            };
+    CeedScalar x[2] = {X[0][i],
+                       X[1][i]
+                      };
     const CeedScalar pidx = X[2][i];
-    // Read dxxdX Jacobian entries, stored in columns
+
+    if (pidx != panel) {
+      const CeedScalar theta  = X[0][i];
+      const CeedScalar lambda = X[1][i];
+
+      CeedScalar T_inv[2][2];
+
+      switch (panel) {
+      case 0:
+      case 1:
+      case 2:
+      case 3:
+        // For P_0 (front), P_1 (east), P_2 (back), P_3 (west):
+        T_inv[0][0] = 1./(cos(theta)*cos(lambda)) * (1./cos(lambda));
+        T_inv[0][1] = 1./(cos(theta)*cos(lambda)) * 0.;
+        T_inv[1][0] = 1./(cos(theta)*cos(lambda)) * tan(theta)*tan(lambda);
+        T_inv[1][1] = 1./(cos(theta)*cos(lambda)) * (1./cos(theta));
+        break;
+      case 4:
+        // For P4 (north):
+        T_inv[0][0] = 1./(sin(theta)*sin(theta)) * sin(theta)*cos(lambda);
+        T_inv[0][1] = 1./(sin(theta)*sin(theta)) * (-sin(lambda));
+        T_inv[1][0] = 1./(sin(theta)*sin(theta)) * sin(theta)*sin(lambda);
+        T_inv[1][1] = 1./(sin(theta)*sin(theta)) * cos(lambda);
+        break;
+      case 5:
+        // For P5 (south):
+        T_inv[0][0] = 1./(sin(theta)*sin(theta)) * (-sin(theta)*cos(lambda));
+        T_inv[0][1] = 1./(sin(theta)*sin(theta)) * sin(lambda);
+        T_inv[1][0] = 1./(sin(theta)*sin(theta)) * sin(theta)*sin(lambda);
+        T_inv[1][1] = 1./(sin(theta)*sin(theta)) * cos(lambda);
+        break;
+      }
+      x[0] = T_inv[0][0]*theta + T_inv[0][1]*lambda;
+      x[1] = T_inv[1][0]*theta + T_inv[1][1]*lambda;
+    }
+
```

Comment:
@jedbrown I implemented the conversion of coordinates from edge (expressed always in lat long) to the different panels. But now it is not clear to me how we relate this to the geometric factors. I think what I did below is wrong. Can we please discuss this? Thanks